### PR TITLE
Run tests with -race

### PR DESCRIPTION
**Pull request by dmathieu**
I had to write this library: https://github.com/dmathieu/safe_buffer
Which introduces a very simple ReadWriter using a mutex.

### CODE REVIEW

File: busltee/transport_test.go [hunk 1]
```diff
@@ -45,7 +47,7 @@ func TestDisconnection(t *testing.T) {
 
 	var callCount int
 	var expectedBody string
-	var stdin bytes.Buffer
+	stdin := safe_buffer.NewMock()
```
Comment (tt) [inline busltee/transport_test.go]: According to [this blog post](https://blog.golang.org/package-names), "good package names are short and clear. They are lower case, with no under_scores or mixedCaps."

Comment (dmathieu) [inline busltee/transport_test.go]: Would you have a name suggestion?

Comment (tt) [inline busltee/transport_test.go]: Just "safebuffer"? Not that different from "httputil" and other packages.

Comment (dmathieu) [inline busltee/transport_test.go]: Sounds good to me. I've renamed the package and rebased this branch.

**Pull request merged**